pydax/_high_level.py

```diff
@@ -0,0 +1,241 @@
+        datasets by calling :func:`pydax.list_all_datasets`.
+    :param version: Version of the dataset to load. Latest version is used by default. You can get a list of all
+        available versions for a dataset by calling :func:`pydax.list_all_datasets`.
+    :param human: Whether to return the metadata as a string in human-readable form or to return a copy of the
+        dataset's schema. Defaults to True.
+    :return: Return a dataset's metadata either as a string or as a schema dictionary.
+    """
+
+    schema_manager = export_schemata()
+    dataset_schema = schema_manager.schemata['datasets'].export_schema('datasets', name, version)
+    license_schema = schema_manager.schemata['licenses'].export_schema('licenses')
+    if human:
+        return (f'Dataset name: {dataset_schema["name"]}\n'
+                f'Description: {dataset_schema["description"]}\n'
+                f'Size: {dataset_schema["estimated_size"]}\n'
+                f'Published date: {dataset_schema["published"]}\n'
+                f'License: {license_schema[dataset_schema["license"]]["name"]}\n'
+                f'Available subdatasets: {", ".join(dataset_schema["subdatasets"].keys())}')
+    else:
+        return dataset_schema
+
+
+def export_schemata() -> SchemaManager:
+    "Return a copy of the ``SchemaManager`` object managed by high-level functions."
+
+    return deepcopy(get_schemata())
+
+
+# The SchemaManager object that is managed by high-level functions
+_schemata: Optional[SchemaManager] = None
+
+
+def load_schemata(*, force_reload: bool = False) -> None:
+    """Loads a :class:`SchemaManager` object that stores all schemata. To export the loaded :class:`SchemaManager`
+    object, please use :func:`.export_schemata`.
+
+    :param force_reload: If ``True``, force reloading even if the provided URLs by :func:`pydax.init` are the same as
+         provided last time. Otherwise, only those that are different from previous given ones are reloaded.
+    """
+    urls = {
+        'datasets': get_config().DATASET_SCHEMA_URL,
+        'formats': get_config().FORMAT_SCHEMA_URL,
+        'licenses': get_config().LICENSE_SCHEMA_URL
+    }
+
+    global _schemata
+    if force_reload or _schemata is None:  # Force reload or clean slate, create a new SchemaManager object
+        _schemata = SchemaManager(**{name: Schema(url) for name, url in urls.items()})
+    else:
+        for name, schema in _schemata.schemata.items():
+            if schema.retrieved_url_or_path != urls[name]:
+                _schemata.add_schema(name, Schema(urls[name]))
+
+
+def get_schemata() -> SchemaManager:
+    """Return the :class:`SchemaManager` object managed by high-level functions. If it is not created, create it. This
+    function is used by high-level APIs but it should not be a high-level function itself. It should only be used
+    internally when the need to modify the managed :class`SchemaManager` object arises. It should not be exposed for the
```

Comment:
@edwardleardi See here

Reply:
My bad, I forgot. What was the reason again for not prepending an underscore to this function? `_handle_version_param` and `_handle_name_param` have underscores despite being in a private module.

Reply:
After the move it should be `_get_schemata` now. It used to be `get_schemata` because other submodules in this module may use these functions, but any PyDAX users shouldn't. But now as the new structure stands, `_get_schemata` should be only used within `_high_level.py`. I've made the changes.

Reply:
ah I see, ok thanks, much better this way :)